jdaviz/core/tests/test_autoconfig.py

```diff
@@ -39,3 +44,43 @@ def test_autoconfig(uris):
 
         assert isinstance(viz_helper, helper_class)
         assert len(viz_helper.app.data_collection) > 0
+
+
+@pytest.mark.remote_data
+@pytest.mark.filterwarnings('ignore')
+def test_launcher():
+    launcher = Launcher(None, ALL_JDAVIZ_CONFIGS)
+
+    # Test starting state
+    assert launcher.hint == STATUS_HINTS['idle']
+    assert launcher.compatible_configs == ALL_JDAVIZ_CONFIGS
+
+    # Test invalid filepath
+    launcher.filepath = "ThisIsAFakeFileNameThatShouldNotExist"
+    assert launcher.hint == STATUS_HINTS['invalid path']
+    assert len(launcher.compatible_configs) == 0
+
+    # Test reseting to empty state
+    launcher.filepath = ""
+    assert launcher.hint == STATUS_HINTS['idle']
+    assert launcher.compatible_configs == ALL_JDAVIZ_CONFIGS
+
+    # Test with real files
+    # Setup temporary directory
+    with TemporaryDirectory(ignore_cleanup_errors=True) as tempdir:
+        for uri, config in AUTOCONFIG_EXAMPLES:
+            if uri.startswith("mast:"):
+                download_path = str(Path(tempdir) / Path(uri).name)
+                Observations.download_file(uri, local_path=download_path)
+            elif uri.startswith("http"):
+                download_path = download_file(uri, cache=True, timeout=100)
+            launcher.filepath = download_path
+            # In testing, setting the filepath will stall until identifying is complete
+            # No need to be concerned for race condition here
+            assert launcher.hint == STATUS_HINTS['id ok']
+            assert launcher.compatible_configs == [config().__class__.__name__.lower()]
```

Comment:
No need to roll your own temp dir handling, use the one pytest gives you.
```suggestion
    for uri, config in AUTOCONFIG_EXAMPLES:
        if uri.startswith("mast:"):
            download_path = str(tmp_path / Path(uri).name)
            Observations.download_file(uri, local_path=download_path)
        elif uri.startswith("http"):
            download_path = download_file(uri, cache=True, timeout=100)
        launcher.filepath = download_path
        # In testing, setting the filepath will stall until identifying is complete
        # No need to be concerned for race condition here
        assert launcher.hint == STATUS_HINTS['id ok']
        assert launcher.compatible_configs == [config().__class__.__name__.lower()]
```